hangman/lib/hangman/impl.ex

```diff
@@ -0,0 +1,124 @@
+defmodule Hangman.Impl do
+ 
+  # state of game
+  defmodule State do
+    defstruct game_state: :initializing, turns_left: 7, letters: [], used: [], last_guess: "", word: "", word_to_list: []
+  end
+
+  # returns a struct representing a new game
+  def new_game() do
+    Dictionary.random_word()
+    |> String.trim("/r")
+    |> initState()
+  end
+
+  # returns the tally for the given game
+  def tally(%State{} = game) do
+    %{
+      game_state: game.game_state,
+      turns_left: game.turns_left,
+      letters:    game.letters,
+      used:       game.used,
+      last_guess: game.last_guess
+    }
+  end
+
+  # returns a tuple containing the updated game state and a tally
+  def make_move(%State{} = game, guess) do
+    game = handle_answer( game, guess, guess in game.used, guess in game.word_to_list )
+    { game, %State{game| last_guess: guess} |> tally() }
+  end
+
+
+
+
+
+  ######################################################################
+  # planed to put this as private function.                            #
+  # However,put it as public function so that we can use it in testing #
+  ######################################################################
+  # game start, initintal the state
+  def initState(inputWord) do
```

Comment:
Your naming here is inconsistent. Most of your functions are (correctly) lower case, but this one is camelCase...

Reply:
Yeah, you are right, that was my bad.
Thank you for remembering me.
Since I used to write lower case as a beginning, but with an upper case at the middle, which was taught by my college professor.(like updateLetter, initeState)
But I noticed your lecture notes are more like update_letter, init_state.
So I write as old style as the beginning of my programming, as I writing with a competition with your lecture notes coding, I became to use your style more.
Later on I will make sure to use the same style(init_state).

Reply:
Not a problem. Different languages have different conventions. All lower-code is Elixir's.